sql/api/src/main/scala/org/apache/spark/sql/streaming/StatefulProcessor.scala

```diff
@@ -60,4 +57,21 @@ private[sql] trait StatefulProcessor[K, I, O] extends Serializable {
    * any cleanup or teardown operations.
    */
   def close (): Unit
+
+  /**
+   * Function to set the stateful processor handle that will be used to interact with the state
+   * store and other stateful processor related operations.
+   * @param handle - instance of StatefulProcessorHandle
+   */
+  final def setStatefulProcessorHandle(handle: StatefulProcessorHandle): Unit = {
+    statefulProcessorHandle = handle
+  }
+
+  /**
+   * Function to get the stateful processor handle that will be used to interact with the state
+   * @return handle - instance of StatefulProcessorHandle
+   */
+  final def getStatefulProcessorHandle: StatefulProcessorHandle = {
+    statefulProcessorHandle
```

Comment:
Lets set the initial value to null. And if its null, then lets raise an exception/error to say that handle is not available ?